File: app/src/main/java/com/cdd/smartsurvey/RegisterPagesNewFamilyActivity.java
package com.cdd.smartsurvey;

import android.Manifest;
import android.annotation.SuppressLint;
import android.app.Activity;
import android.app.DatePickerDialog;
import android.content.DialogInterface;
import android.content.Intent;
import android.content.pm.ActivityInfo;
import android.graphics.Bitmap;
import android.graphics.Color;
import android.graphics.drawable.ColorDrawable;
import android.os.Bundle;
import android.provider.MediaStore;
import android.text.Editable;
import android.text.TextWatcher;
import android.view.View;
import android.widget.AdapterView;
import android.widget.Button;
import android.widget.EditText;
import android.widget.ListView;
import android.widget.SimpleAdapter;
import android.widget.TextView;
import android.widget.Toast;

import androidx.appcompat.app.AlertDialog;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.app.ActivityCompat;

import com.android.volley.RequestQueue;
import com.android.volley.toolbox.Volley;
import com.cdd.smartsurvey.sqlite.DatabaseHelper;
import com.cdd.smartsurvey.sqlite.model.Gender;
import com.cdd.smartsurvey.sqlite.model.Prefix;
import com.cdd.smartsurvey.utils.AllMessage;
import com.cdd.smartsurvey.utils.GetThaiCharector;
import com.cdd.smartsurvey.utils.ImageUtil;
import com.microblink.MicroblinkSDK;
import com.microblink.entities.recognizers.RecognizerBundle;
import com.microblink.entities.recognizers.blinkid.generic.BlinkIdCombinedRecognizer;
import com.microblink.intent.IntentDataTransferMode;
import com.microblink.uisettings.ActivityRunner;
import com.microblink.uisettings.BlinkIdUISettings;

import org.json.JSONObject;

import java.util.ArrayList;
import java.util.Calendar;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

import de.hdodenhof.circleimageview.CircleImageView;

public class RegisterPagesNewFamilyActivity extends AppCompatActivity {

    private static final String TAG = "registerPagesActivity";
    public static final int MY_BLINKID_REQUEST_CODE = 123;

    public Button btnCamera;
    public Button btnCaptureID;
    public Button btnClose;
    public Button btnCloseMain;
    public Button btnSave;
    public EditText txtName;
    public EditText txtSurname;
    public EditText txtAddress;
    public EditText txtCardID;
    public TextView txtBirthDate;
    public TextView txtPrefix;
    public TextView txtGender;
    public String   txtPicture;
    public JSONObject jobj;
    public RequestQueue MyRequestQueue;
    public String   returnOut;
    public AllMessage allMessage;

    CircleImageView imageView;

    private TextView mDisplayDate;
    private DatePickerDialog.OnDateSetListener mDateSetListener;
    private BlinkIdCombinedRecognizer recognizer;
    private RecognizerBundle recognizerBundle;
    public Button btnStartSurvey;
    public Button btnBack;

    @SuppressLint("SourceLockedOrientationActivity")
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        MicroblinkSDK.setLicenseFile(GlobalValue.LicenseFile,this);
        MicroblinkSDK.setIntentDataTransferMode(IntentDataTransferMode.PERSISTED_OPTIMISED);
        setContentView(R.layout.activity_register_newfamily);
        setRequestedOrientation(ActivityInfo.SCREEN_ORIENTATION_PORTRAIT);

        ActivityCompat.requestPermissions(this,
                new String[]{Manifest.permission.CAMERA},0);
        recognizer = new BlinkIdCombinedRecognizer();
        recognizerBundle = new RecognizerBundle(recognizer);
        allMessage = new AllMessage();
        MyRequestQueue = Volley.newRequestQueue(this);

        btnCamera = (Button)findViewById(R.id.btnCapture);
        imageView = (CircleImageView)findViewById(R.id.profileimage);

        btnCamera.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view){
                Intent intent = new Intent (MediaStore.ACTION_IMAGE_CAPTURE);
                startActivityForResult(intent,1);
            }
        });

        mDisplayDate = (TextView) findViewById(R.id.txtBirthDate);
        mDisplayDate.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Calendar cal = Calendar.getInstance();
                int year  = cal.get(Calendar.YEAR);
                int month = cal.get(Calendar.MONTH);
                int day   = cal.get(Calendar.DAY_OF_MONTH);

                DatePickerDialog dialog = new DatePickerDialog(
                        RegisterPagesNewFamilyActivity.this,
                        android.R.style.Theme_Holo_Light_Dialog_MinWidth,
                        mDateSetListener,
                        year,month,day);
                dialog.getWindow().setBackgroundDrawable(new ColorDrawable(Color.TRANSPARENT));
                dialog.show();
            }
        });

        btnCaptureID = (Button)findViewById(R.id.btnCaptureID);
        btnCaptureID.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view){
                openCaptureID();
            }
        });

        btnSave = (Button) findViewById(R.id.btnSave);
        btnSave.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view){
                SendData(GlobalValue.dbUrl);
            }
        });

        txtName = (EditText) findViewById(R.id.txtName);
        txtName.addTextChangedListener(new TextWatcher() {
            @Override
            public void onTextChanged(CharSequence s, int start, int before, int count) {
                try {
                    if (!GetThaiCharector.checkThai(s))
                    {
                        AlertDialog.Builder ab = new AlertDialog.Builder(RegisterPagesNewFamilyActivity.this, R.style.AlertDialogTheme);
                        ab.setTitle("แจ้งเตือน");
                        ab.setMessage("ต้องใช้เป็นภาษาไทยเท่านั้น");
                        ab.setPositiveButton("ตกลง", new DialogInterface.OnClickListener() {
                            @Override
                            public void onClick(DialogInterface dialog, int which) {
                                dialog.dismiss();
                                txtName.setText("");
                            }
                        });
                        ab.show();
                    }
                }
                catch (Exception e){
                }
            }
            @Override
            public void beforeTextChanged(CharSequence s, int start, int count, int after) {
                // TODO Auto-generated method stub
            }
            @Override
            public void afterTextChanged(Editable s) {
                // TODO Auto-generated method stub
            }
        });

        txtSurname = (EditText)findViewById(R.id.txtSurname);
        txtSurname.addTextChangedListener(new TextWatcher() {
            @Override
            public void onTextChanged(CharSequence s, int start, int before, int count) {
                try {
                    if (!GetThaiCharector.checkThai(s))
                    {
                        AlertDialog.Builder ab = new AlertDialog.Builder(RegisterPagesNewFamilyActivity.this, R.style.AlertDialogTheme);
                        ab.setTitle("แจ้งเตือน");
                        ab.setMessage("ต้องใช้เป็นภาษาไทยเท่านั้น");
                        ab.setPositiveButton("ตกลง", new DialogInterface.OnClickListener() {
                            @Override
                            public void onClick(DialogInterface dialog, int which) {
                                dialog.dismiss();
                            }
                        });
                        ab.show();
                    }
                }
                catch (Exception e){
                }
            }
            @Override
            public void beforeTextChanged(CharSequence s, int start, int count, int after) {
                // TODO Auto-generated method stub
            }
            @Override
            public void afterTextChanged(Editable s) {
                // TODO Auto-generated method stub
            }
        });

        txtBirthDate = (TextView) findViewById(R.id.txtBirthDate);

        txtPrefix = (TextView) findViewById(R.id.txtPrefix);
        txtPrefix.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                ShowAlertDialogWithPrefix_Listview();
            }
        });

        txtGender = (TextView) findViewById(R.id.txtGender);
        txtGender.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                ShowAlertDialogWithGender_Listview();
            }
        });


        btnBack = (Button)findViewById(R.id.btnBack);
        btnBack.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view){
                onBackPressed();
            }
        });
    }

    @Override
    protected void onActivityResult(int requestCode, int resultCode, Intent data){
        super.onActivityResult(requestCode,resultCode,data);

        try {

            Bitmap bitmap = (Bitmap)data.getExtras().get("data");
            if (bitmap != null) {
                txtPicture = ImageUtil.convert(bitmap);
                imageView.setImageBitmap(bitmap);
            }

            if (requestCode != MY_BLINKID_REQUEST_CODE) {
                return;
            }

            if (resultCode == Activity.RESULT_OK) {
                onScanSuccess(data);
            } else {
                onScanCanceled();
            }

        } catch (Exception e) {
            e.printStackTrace();
        }
    }

    private void onScanSuccess(Intent data) {
        recognizerBundle.loadFromIntent(data);

        BlinkIdCombinedRecognizer.Result result = recognizer.getResult();
        String birthDate = result.getDateOfBirth().getDate().toString().substring(0,10).replace('.','/');
        String cardID = result.getDocumentNumber();
        if (cardID.isEmpty()) {
            cardID = result.getDocumentNumber();
        }
        txtCardID = (EditText) findViewById(R.id.txtCardID);
        txtCardID.setText(cardID);

        txtBirthDate = (TextView) findViewById(R.id.txtBirthDate);
        txtBirthDate.setText(birthDate);
    }

    private void onScanCanceled() {
        Toast.makeText(this, "Scan cancelled!", Toast.LENGTH_SHORT).show();
    }

    public void openMenuDetailSurvey() {
        Intent intent = new Intent(this,MenuSurveyActivity.class);
        startActivity(intent);
    }

    public void openCaptureID() {
        BlinkIdUISettings uiSettings = new BlinkIdUISettings(recognizerBundle);
        ActivityRunner.startActivityForResult(this, MY_BLINKID_REQUEST_CODE, uiSettings);
    }

    public void ShowAlertDialogWithPrefix_Listview()
    {
        EditText txtSearch;
        DatabaseHelper db;

        final List<Prefix> prefixList = new ArrayList<>();


        db = new DatabaseHelper(this);

        prefixList.addAll(db.getAllPrefixs());
        Prefix item;

        final ArrayList<Map<String,Object>> itemDataList = new ArrayList<Map<String,Object>>();;

        for(int i=0;i<prefixList.size();i++) {
            item = prefixList.get(i);
            Map<String,Object> listItemMap = new HashMap<String,Object>();
            listItemMap.put("code", item.getCode());
            listItemMap.put("codename", item.getCodename());
            itemDataList.add(listItemMap);
        }

        final AlertDialog.Builder mBuilder = new AlertDialog.Builder(RegisterPagesNewFamilyActivity.this);
        final View mView = getLayoutInflater().inflate(R.layout.dialog_prefix_valuelist, null);

        final SimpleAdapter prefixAdapter = new SimpleAdapter(this,itemDataList,android.R.layout.simple_list_item_2,
                new String[]{"codename"},new int[]{android.R.id.text1});

        txtSearch = (EditText) mView.findViewById(R.id.editText);
        txtSearch.addTextChangedListener(new TextWatcher() {
            @Override
            public void beforeTextChanged(CharSequence s, int start, int count, int after) {

            }

            @Override
            public void onTextChanged(CharSequence s, int start, int before, int count) {
                prefixAdapter.getFilter().filter(s);
            }

            @Override
            public void afterTextChanged(Editable s) {

            }
        });

        ListView list = (ListView) mView.findViewById(R.id.lstValue);
        list.setAdapter(prefixAdapter);

        mBuilder.setView(mView);
        final AlertDialog show = mBuilder.show();

        list.setOnItemClickListener(new AdapterView.OnItemClickListener() {
            @Override
            public void onItemClick(AdapterView<?> adapterView, View view, int index, long l) {
                Object clickItemObj = adapterView.getAdapter().getItem(index);
                HashMap clickItemMap = (HashMap)clickItemObj;
                txtPrefix.setTag(clickItemMap.get("code"));
                txtPrefix.setText((String)clickItemMap.get("codename"));
                if (txtPrefix.getTag().equals("001") || txtPrefix.getTag().equals("002")) {
                    txtGender.setTag("1");
                    txtGender.setText("ชาย (male)");
                }else{
                    txtGender.setTag("2");
                    txtGender.setText("หญิง (Female)");
                }
                show.dismiss();
            }
        });

        btnClose = (Button) mView.findViewById(R.id.btnClose);
        btnClose.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view){
                show.dismiss();
            }
        });
    }

    public void ShowAlertDialogWithGender_Listview()
    {
        EditText txtSearch;
        List<Gender> genderList = new ArrayList<>();

        DatabaseHelper db;

        db = new DatabaseHelper(this);

        genderList.addAll(db.getAllGenders());
        Gender item;

        ArrayList<Map<String,Object>> itemDataList = new ArrayList<Map<String,Object>>();;

        for(int i=0;i<genderList.size();i++) {
            item = genderList.get(i);
            Map<String,Object> listItemMap = new HashMap<String,Object>();
            listItemMap.put("code", item.getCode());
            listItemMap.put("codename", item.getCodename());
            itemDataList.add(listItemMap);
        }

        final AlertDialog.Builder mBuilder = new AlertDialog.Builder(RegisterPagesNewFamilyActivity.this);
        View mView = getLayoutInflater().inflate(R.layout.dialog_gender_valuelist, null);

        final SimpleAdapter genderAdapter = new SimpleAdapter(this,itemDataList,android.R.layout.simple_list_item_2,
                new String[]{"codename"},new int[]{android.R.id.text1});

        txtSearch = (EditText) mView.findViewById(R.id.editText);
        txtSearch.addTextChangedListener(new TextWatcher() {
            @Override
            public void beforeTextChanged(CharSequence s, int start, int count, int after) {

            }

            @Override
            public void onTextChanged(CharSequence s, int start, int before, int count) {
                genderAdapter.getFilter().filter(s);
            }

            @Override
            public void afterTextChanged(Editable s) {

            }
        });

        ListView list = (ListView) mView.findViewById(R.id.lstValue);
        list.setAdapter(genderAdapter);

        mBuilder.setView(mView);
        final AlertDialog show = mBuilder.show();

        list.setOnItemClickListener(new AdapterView.OnItemClickListener() {
            @Override
            public void onItemClick(AdapterView<?> adapterView, View view, int index, long l) {
                Object clickItemObj = adapterView.getAdapter().getItem(index);
                HashMap clickItemMap = (HashMap)clickItemObj;
                txtGender.setTag(clickItemMap.get("code"));
                txtGender.setText((String)clickItemMap.get("codename"));
                show.dismiss();
            }
        });

        btnClose = (Button) mView.findViewById(R.id.btnClose);
        btnClose.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view){
                show.dismiss();
            }
        });
    }

    public void SendData (String aurl) {
        Intent intent = new Intent(this,AddressSurveyActivity.class);
        startActivity(intent);
    }

}

File: app/src/main/java/com/cdd/smartsurvey/RegisterPagesNewFamilyActivity.kt
package com.cdd.smartsurvey

import android.Manifest
import android.annotation.SuppressLint
import android.content.Intent
import android.content.pm.ActivityInfo
import android.graphics.Bitmap
import android.graphics.BitmapFactory
import android.net.Uri
import android.os.Bundle
import android.os.Environment
import android.provider.MediaStore
import android.text.Editable
import android.text.TextWatcher
import android.util.Log
import android.view.View
import android.widget.*
import androidx.appcompat.app.AlertDialog
import androidx.appcompat.app.AppCompatActivity
import androidx.core.app.ActivityCompat
import androidx.core.content.FileProvider
import com.cdd.smartsurvey.data.model.Family
import com.cdd.smartsurvey.http.model.Community
import com.cdd.smartsurvey.utils.*
import com.layernet.thaidatetimepicker.date.DatePickerDialog
import de.hdodenhof.circleimageview.CircleImageView
import kotlinx.android.synthetic.main.activity_register_newfamily.*
import java.io.ByteArrayOutputStream
import java.io.File
import java.io.FileOutputStream
import java.io.IOException
import java.text.SimpleDateFormat
import java.util.*
import kotlin.collections.ArrayList
import kotlin.collections.HashMap

class RegisterPagesNewFamilyActivity : AppCompatActivity(), DatePickerDialog.OnDateSetListener {
    companion object {
        private const val TAG = "registerPagesActivity"
    }

    private val REQUEST_IMAGE_CAPTURE = 1
    private val REQUEST_TAKE_PHOTO = 1

    lateinit var currentPhotoPath: String
    var allMessage: AllMessage? = null
    var imageView: CircleImageView? = null

    @SuppressLint("SourceLockedOrientationActivity")
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_register_newfamily)
        requestedOrientation = ActivityInfo.SCREEN_ORIENTATION_PORTRAIT
        ActivityCompat.requestPermissions(this, arrayOf(Manifest.permission.CAMERA), 0)
        allMessage = AllMessage()
        imageView = findViewById<View>(R.id.profileimage) as CircleImageView
        btnCapture!!.setOnClickListener {
            val intent = Intent(MediaStore.ACTION_IMAGE_CAPTURE)
            startActivityForResult(intent, 1)
        }

        txtBirthDate.setOnClickListener {
            val now = Calendar.getInstance()
            val dpd = DatePickerDialog.newInstance(
                    this@RegisterPagesNewFamilyActivity,
                    now[Calendar.YEAR],
                    now[Calendar.MONTH],
                    now[Calendar.DAY_OF_MONTH]
            )

            dpd.show(fragmentManager, "Datepickerdialog");
        }
        btnSave.setOnClickListener { StoreData() }
        txtName.addTextChangedListener(object : TextWatcher {
            override fun onTextChanged(s: CharSequence, start: Int, before: Int, count: Int) {
                try {
                    if (!GetThaiCharector.checkThai(s)) {
                        val ab = AlertDialog.Builder(this@RegisterPagesNewFamilyActivity, R.style.AlertDialogTheme)
                        ab.setTitle("แจ้งเตือน")
                        ab.setMessage("ต้องใช้เป็นภาษาไทยเท่านั้น")
                        ab.setPositiveButton("ตกลง") { dialog, which ->
                            dialog.dismiss()
                            txtName!!.setText("")
                        }
                        ab.show()
                    }
                } catch (e: Exception) {
                }
            }

            override fun beforeTextChanged(s: CharSequence, start: Int, count: Int, after: Int) {
                // TODO Auto-generated method stub
            }

            override fun afterTextChanged(s: Editable) {
                // TODO Auto-generated method stub
            }
        })
        txtSurname.addTextChangedListener(object : TextWatcher {
            override fun onTextChanged(s: CharSequence, start: Int, before: Int, count: Int) {
                try {
                    if (!GetThaiCharector.checkThai(s)) {
                        val ab = AlertDialog.Builder(this@RegisterPagesNewFamilyActivity, R.style.AlertDialogTheme)
                        ab.setTitle("แจ้งเตือน")
                        ab.setMessage("ต้องใช้เป็นภาษาไทยเท่านั้น")
                        ab.setPositiveButton("ตกลง") { dialog, which -> dialog.dismiss() }
                        ab.show()
                    }
                } catch (e: Exception) {
                }
            }

            override fun beforeTextChanged(s: CharSequence, start: Int, count: Int, after: Int) {
                // TODO Auto-generated method stub
            }

            override fun afterTextChanged(s: Editable) {
                // TODO Auto-generated method stub
            }
        })
        val formUtil = FormUtils(this@RegisterPagesNewFamilyActivity, layoutInflater)
        txtPrefix.setOnClickListener { formUtil.ShowAlertDialogWithPrefix_Listview(txtPrefix, txtGender) }
        txtGender.setOnClickListener { formUtil.ShowAlertDialogWithGender_Listview(txtGender) }
        btnBack.setOnClickListener { onBackPressed() }
    }

    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
        super.onActivityResult(requestCode, resultCode, data)
        if (requestCode == REQUEST_IMAGE_CAPTURE && resultCode == RESULT_OK) {
            val processBitmap = resizeBitmap(currentPhotoPath)
            saveImage(processBitmap, currentPhotoPath)
//            val processBitmap = BitmapFactory.decodeFile(currentPhotoPath)
            profileimage.setImageBitmap(processBitmap)
        }

//        try {
//            val bitmap = data!!.extras["data"] as Bitmap
//            if (bitmap != null) {
//                txtPicture = ImageUtil.convert(bitmap)
//                imageView!!.setImageBitmap(bitmap)
//            }
//            if (requestCode != MY_BLINKID_REQUEST_CODE) {
//                return
//            }
//        } catch (e: Exception) {
//            e.printStackTrace()
//        }
    }

    fun openMenuDetailSurvey() {
        val intent = Intent(this, MenuSurveyActivity::class.java)
        startActivity(intent)
    }

    fun StoreData() {
        val selectCommunity: Community = intent.getParcelableExtra(GlobalValue.EXTRA_COMMUNITY)

        var family = Family()

        with(family) {
            community = selectCommunity.community_id
            prefix = txtPrefix.tag.toString()
            fname = txtName.text.toString()
            lname = txtSurname.text.toString()
            idcard = txtCardID.text.toString()
            gender = txtGender.tag.toString()
            birthdate = txtBirthDate.text.toString()
        }


        val intent = Intent(this, AddressSurveyActivity::class.java).apply {
            putExtra(GlobalValue.EXTRA_COMMUNITY, selectCommunity)
            putExtra(GlobalValue.EXTRA_FAMILY, family)
        }
        startActivity(intent)
    }

    private fun dispatchTakePictureIntent() {
        Intent(MediaStore.ACTION_IMAGE_CAPTURE).also { takePictureIntent ->
            // Ensure that there's a camera activity to handle the intent
            takePictureIntent.resolveActivity(packageManager)?.also {
                // Create the File where the photo should go
                val photoFile: File? = try {
                    createImageFile()
                } catch (ex: IOException) {
                    // Error occurred while creating the File
                    // Error occurred while creating the File

                    null
                    null
                }
                // Continue only if the File was successfully created
                photoFile?.also {
                    val photoURI: Uri = FileProvider.getUriForFile(
                            this,
                            "com.cdd.smartsurvey.fileprovider",
                            it
                    )
                    takePictureIntent.putExtra(MediaStore.EXTRA_OUTPUT, photoURI)
                    startActivityForResult(takePictureIntent, REQUEST_TAKE_PHOTO)
                }
            }

        }
    }

    @Throws(IOException::class)
    private fun createImageFile(): File {
        // Create an image file name
        val timeStamp: String = SimpleDateFormat("yyyyMMdd_HHmmss").format(Date())
        val storageDir: File = getExternalFilesDir(Environment.DIRECTORY_PICTURES)
        return File.createTempFile(
                "JPEG_${timeStamp}_", /* prefix */
                ".jpg", /* suffix */
                storageDir /* directory */
        ).apply {
            // Save a file: path for use with ACTION_VIEW intents
            currentPhotoPath = absolutePath
        }
    }

    fun resizeBitmap(imagePath: String): Bitmap {
        val scale = resources.getDisplayMetrics().density
        var b = BitmapFactory.decodeFile(imagePath)

        val origWidth = b.width
        val origHeight = b.height

        val destWidth = 2048

        if (origWidth > destWidth) {
            // picture is wider than we want it, we calculate its target height
            val destHeight = origHeight / (origWidth / destWidth);
            // we create an scaled bitmap so it reduces the image, not just trim it
            var b2 = Bitmap.createScaledBitmap(b, destWidth, destHeight, false);

            return b2
        }

        return b
    }

    private fun saveImage(finalBitmap: Bitmap, toPath: String) {
        var bos = ByteArrayOutputStream()
        finalBitmap.compress(Bitmap.CompressFormat.JPEG, 90, bos)
        var data = bos.toByteArray()

        val file = File(toPath)
        if (file.exists()) file.delete()
        try {
            val stream = FileOutputStream(file)
            stream.write(data)
        } catch (e: Exception) {
            e.printStackTrace()
        }

    }

    override fun onDateSet(view: DatePickerDialog?, year: Int, monthOfYear: Int, dayOfMonth: Int) {
        Log.d(RegisterPagesNewFamilyActivity.TAG, "onDateSet: dd/mm/yyyy: $dayOfMonth/${monthOfYear + 1}/${year + 543}")
        val date = "$dayOfMonth/${monthOfYear + 1}/${year + 543}"
        txtBirthDate.text = date
    }
}